app/core/Engine.js

```diff
@@ -64,70 +64,122 @@ class Engine {
+				infuraProjectId: process.env.MM_INFURA_PROJECT_ID || EMPTY,
+				providerConfig: {
+					static: {
+						eth_sendTransaction: async (payload, next, end) => {
+							const { TransactionController } = this.context;
+							try {
+								const hash = await (await TransactionController.addTransaction(
+									payload.params[0],
+									payload.origin
+								)).result;
+								end(undefined, hash);
+							} catch (error) {
+								end(error);
 							}
 						}
-					}),
-					new PhishingController(),
-					new PreferencesController(
-						{},
-						{
-							ipfsGateway: AppConstants.IPFS_DEFAULT_GATEWAY_URL
-						}
+					},
+					getAccounts: (end, payload) => {
+						const { approvedHosts, privacyMode } = store.getState();
+						const isEnabled = !privacyMode || approvedHosts[payload.hostname];
+						const { KeyringController } = this.context;
+						const isUnlocked = KeyringController.isUnlocked();
+						const selectedAddress = this.context.PreferencesController.state.selectedAddress;
+						end(null, isUnlocked && isEnabled && selectedAddress ? [selectedAddress] : []);
+					}
+				}
+			});
+			const assetsContractController = new AssetsContractController();
+			const assetsController = new AssetsController({
+				onPreferencesStateChange: listener => preferencesController.subscribe(listener),
+				onNetworkStateChange: listener => networkController.subscribe(listener),
+				getAssetName: assetsContractController.getAssetName.bind(assetsContractController),
+				getAssetSymbol: assetsContractController.getAssetSymbol.bind(assetsContractController),
+				getCollectibleTokenURI: assetsContractController.getCollectibleTokenURI.bind(assetsContractController)
+			});
+			const currencyRateController = new CurrencyRateController({
+				nativeCurrency,
+				currentCurrency
+			});
+
+			const controllers = [
+				new KeyringController(
+					{
+						removeIdentity: preferencesController.removeIdentity.bind(preferencesController),
+						syncIdentities: preferencesController.syncIdentities.bind(preferencesController),
+						updateIdentities: preferencesController.updateIdentities.bind(preferencesController),
+						setSelectedAddress: preferencesController.setSelectedAddress.bind(preferencesController)
+					},
+					{ encryptor },
+					initialState.KeyringController
+				),
+				new AccountTrackerController({
+					onPreferencesStateChange: listener => preferencesController.subscribe(listener),
+					initialIdentities: initialState.preferencesController?.identities
+				}),
+				new AddressBookController(),
+				assetsContractController,
+				assetsController,
+				new AssetsDetectionController({
+					onAssetsStateChange: listener => assetsController.subscribe(listener),
+					onPreferencesStateChange: listener => preferencesController.subscribe(listener),
+					onNetworkStateChange: listener => networkController.subscribe(listener),
+					getOpenSeaApiKey: assetsController.openSeaApiKey,
```

Comment:
I was having issues with this but it worked with 

```suggestion
					getOpenSeaApiKey: () => assetsController.openSeaApiKey,
```

Reply:
Ah, of course! :facepalm: Good catch!